src/routes/fastPayment.js

```diff
@@ -18,18 +19,24 @@ route.post('/fastPayment',
         const price = req.body.price;
         const filter = { userID: req.body.userID };
 
-        userDB.findOne(filter, (err, user) => {
-            if (err) {
-                console.log(err);
-                res.status(500);
-                res.send("Error Has Occured");
-            } else {
-                const result = makePayment(price, user);
-                res.json({ user }); //test purposes!
+        const user = await userDB.findOne(filter);
 
+        try {
+            const iyzicoResult = await makePayment(price, user);
+            const paymentInfo = {
+                paymentID: iyzicoResult.paymentId,
+                status: iyzicoResult.status
             }
-
-        });
+            if (iyzicoResult.status === 'success') {
+                const payment = await paymentDB.findOneAndUpdate(filter, paymentInfo, { new: true });
+                res.json({ payment });
+            } else {
+                const payment = await paymentDB.findOneAndUpdate(filter, paymentInfo, { new: true });
+                res.json({ payment, iyzicoResult });
```

Comment:
return the same structure always, line 32 and line 35 returns different structures.